node/rpcstack.go

```diff
@@ -257,8 +277,25 @@ func validatePrefix(what, path string) error {
 // stop shuts down the HTTP server.
 func (h *httpServer) stop() {
 	h.mu.Lock()
-	defer h.mu.Unlock()
-	h.doStop()
+	h.ready = false
+	time.AfterFunc(stopPendingRequestTimeout, func() {
+		defer h.mu.Unlock()
+		h.doStop()
+		if h.shutdown != nil {
+			func() {
+				// Avoid panic when closing closed channel
+				defer func() { recover() }()
+				close(h.shutdown)
+			}()
+		}
+	})
+}
+
+// ShutdownWait waits for the server to shutdown.
```

Comment:
nitpick: This function is not called `ShutdownWait`.
```suggestion
// wait waits for the server to shutdown.
```